src/main/kotlin/cacophony/codegen/instructions/cacophonyInstructions/ArithmeticInstructions.kt

```diff
@@ -89,6 +89,9 @@ data class IDiv(val reg: Register) : Instruction {
 
     override fun toAsm(hardwareRegisterMapping: HardwareRegisterMapping): String {
         val hardwareReg = hardwareRegisterMapping[reg]
+
+        require(hardwareReg != null) { "No hardware register mapping found for $reg" }
```

Comment:
nice idea, I was just thinking if we could do anything to avoid that much code repetition

Reply:
i'm planning to do some refactoring during holidays, also in these parts of code
maybe I'll come up with something clever then